### DISCUSSION TITLE
Weather interface adjustment announcement 天气接口调整公告

### DISCUSSION THREAD

**Discussion by MyoungXUE** [zepp-health/zeppos-forum#83] [Announcements]
# Weather interface adjustment announcement

## Adjustment background

Due to the platform's business restructuring, weather-related interfaces will be deprecated, while we encourage developers to develop more flexible and dynamic weather applications using the capabilities of web requests.

## Adjustment Description

As of March 31, 2023 at 24:00, the [Weather sensor interface](https://docs.zepp.com/docs/reference/device-app-api/newAPI/sensor/Weather/) will be marked as deprecated and the associated data return value will be adjusted to `undefined`.

**Added on February 23, 2023, update the scope of the impact of adjusting the interface, only the API related to the applet is affected, the dial API is not affected.**

## Adjustment suggestions

It is recommended that developers use [FetchAPI](https://docs.zepp.com/docs/reference/side-service-api/fetch/) to obtain relevant weather data via web requests.

Refer to the following code example

```js
// side-service
fetch("https://exampleweatherapi.com/weather/now?location=xxx")
  .then((data) => {
    console.log(data)
  })
```

Then use the message related API to communicate between Device App and Side Service.

For the full FetchAPI usage, please refer to https://docs.zepp.com/docs/samples/app/fetchAPI/

Thank you for your support and attention to the Zepp OS.

Zepp OS Team
February 20, 2023

---
# 天气接口调整公告

## 调整背景

因平台业务调整，天气相关接口将做废弃处理，同时我们鼓励开发者使用网络请求的能力开发更灵活且动态的天气类应用程序。

## 调整说明

自 2023 年 3 月 31 日 24 时后，[Weather 传感器接口](https://docs.zepp.com/zh-cn/docs/reference/device-app-api/newAPI/sensor/Weather/)将标记为废弃，相关数据返回值将会为调整为 `undefined`。

**2023 年 2 月 23 日补充，更新调整接口影响范围，仅小程序相关 API 受影响，表盘 API 不受影响。**

## 调整建议

建议开发者使用 [FetchAPI](https://docs.zepp.com/zh-cn/docs/reference/side-service-api/fetch/) 通过网络请求获取相关天气数据。

参考以下代码示例：

```js
// side-service
fetch("https://exampleweatherapi.com/weather/now?location=xxx")
  .then((data) => {
    console.log(data)
  })
```

再使用 message 相关 API 做到 「设备应用」 和 「伴生服务」 之间的通信即可。

完整 FetchAPI 的使用请参考 https://docs.zepp.com/zh-cn/docs/reference/side-service-api/fetch/

感谢您对平台的支持与关注！

Zepp OS 团队
2023 年 2 月 20 日

**Comment by mimimix** [comment]
What about the watchface?

**Reply by MyoungXUE** [reply]
> What about the watchface?

Hi, we have added the watchface related section in the announcement, the watchface API is not affected by this adjustment.
